slushfile.js

```diff
@@ -238,6 +242,42 @@ gulp.task('configEsri', ['init'], function(done) {
   done();
 });
 
+gulp.task('configGulp', ['init'], function(done) {
+
+  try {
+
+    var config = fs.readFileSync('gulp.config.js', { encoding: 'utf8' });
+
+    //set the ports
+    config = config.replace(/\bdefaultPort: '9070'/m, 'defaultPort: \'' + settings.nodePort + '\'');
+    config = config.replace(/\bport: '8040'/m, 'port: \'' + settings.appPort + '\'');
+
+    fs.writeFileSync('gulp.config.js', config);
+  } catch (e) {
+    console.log('failed to update gulp.config.js: ' + e.message);
+  }
+
+  done();
+});
+
+gulp.task('configRoutes', ['init'], function(done) {
+
+  try {
+
+    var routes = fs.readFileSync('node-server/routes.js', { encoding: 'utf8' });
+
+    //set the ports
+    routes = routes.replace(/\bprocess.env.APP_PORT \|\| 9070/m, 'process.env.APP_PORT || \'' + settings.nodePort + '\'');
```

Comment:
One final change requested @hunterwilliams .

Rather than replacing the lines in routes.js, could you just modify routes.js to read like proxy.js:

```javascript
var config = require('../gulp.config')();

var options = {
  appPort: process.env.APP_PORT || config.defaultPort,
  mlHost: process.env.ML_HOST || config.marklogic.host,
  mlPort: process.env.ML_PORT || config.marklogic.port,
  defaultUser: config.marklogic.user,
  defaultPass: config.marklogic.password
};
```

Then you don't need to replace anything. It will just work because of your configGulp task. This is my bad for not making them the same when I refactored the code base.